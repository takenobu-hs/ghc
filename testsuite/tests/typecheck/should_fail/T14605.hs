{-# Language TypeApplications #-}
{-# Language ImpredicativeTypes #-}
-- This isn't a test for impredicative types; it's
-- just that visible type application on a for-all type
-- is an easy way to provoke the error.
--
-- The ticket #14605 has a much longer example that
-- also fails; it does not use ImpredicativeTypes

module T14605 where

import GHC.Prim (coerce)

duplicate = coerce @(forall x. ()) @(forall x. x)
