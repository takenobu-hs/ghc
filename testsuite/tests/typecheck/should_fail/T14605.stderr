
T14605.hs:14:13: error:
    • Couldn't match representation of type ‘x1’ with that of ‘()’
        arising from a use of ‘coerce’
      ‘x1’ is a rigid type variable bound by
        the type ()
        at T14605.hs:14:1-49
    • In the expression: coerce @(forall x. ()) @(forall x. x)
      In an equation for ‘duplicate’:
          duplicate = coerce @(forall x. ()) @(forall x. x)
